ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -263,4 +263,25 @@ - (void)testShouldHandleVnodeOpEvent {
     XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
 }
 
+//KEXT_STATIC void UseMainForkIfNamedStream(vnode_t& vnode, bool& putVnodeWhenDone);
+- (void)testUseMainForkIfNamedStream {
+    shared_ptr<mount> testMount = mount::Create();
+    const char* filePath = "/Users/test/code/Repo/file";
+    shared_ptr<vnode> testFileVnode = testMount->CreateVnodeTree(filePath);
+
+    // Out Parameters to test
+    bool result;
+    vnode_t testVnode = testFileVnode.get();
+
+    testVnode->namedStream = false;
+    UseMainForkIfNamedStream(testVnode, result);
+    XCTAssertFalse(result);
+    
+    testVnode->namedStream = true;
+    UseMainForkIfNamedStream(testVnode, result);
```

Comment:
I think this test is good for now, but I'd be interested to hear if @pmj has any ideas on how to better mock named streams.

As it is now this call to `UseMainForkIfNamedStream` returns the parent directory vnode ("/Users/test/code/Repo") while in production I would not expect `UseMainForkIfNamedStream` to ever return a directory vnode.

Reply:
Under what circumstances would `UseMainForkIfNamedStream `.  Would a link do it?

Reply:
I believe that on the real file system this would happen for named/resource forks (I saw a discussion on how to create them here: https://apple.stackexchange.com/questions/228444/how-do-i-create-a-named-fork-and-store-data-in-it).  

There were also a few (somewhat vague) details here: https://support.apple.com/en-us/HT202021

@nickgra, do you have any tricks for created files that use named streams?